src/Servers/Kestrel/Core/src/Internal/Http2/Http2OutputProducer.cs

```diff
@@ -449,6 +455,12 @@ private static void ThrowSuffixSent()
             throw new InvalidOperationException("Writing is not allowed after writer was completed.");
         }
 
+        [StackTraceHidden]
+        private static void ThrowDisposed()
+        {
+            throw new InvalidOperationException("Cannot write to response after the request middleware has completed.");
```

Comment:
nit: "Cannot write to response after the request has completed." No need to mention middleware.